packages/core/strapi/lib/services/entity-service/__tests__/entity-service.test.js

```diff
@@ -79,16 +82,46 @@ describe('Entity service', () => {
   });
 
   describe('Create', () => {
-    describe('assign default values', () => {
-      let instance;
+    let instance;
+    const fakeQuery = {
+      count: jest.fn(() => 0),
+      create: jest.fn(({ data }) => ({
+        id: 1,
+        ...data,
+      })),
+      findOne: jest.fn(),
+    };
+    const fakeModels = {};
 
-      beforeAll(() => {
-        const fakeQuery = {
-          count: jest.fn(() => 0),
-          create: jest.fn(({ data }) => data),
-        };
+    beforeAll(() => {
+      global.strapi.getModel.mockImplementation((modelName) => fakeModels[modelName]);
+      global.strapi.query.mockImplementation(() => fakeQuery);
 
-        const fakeModel = {
+      const fakeDB = {
+        query: jest.fn(() => fakeQuery),
+      };
+
+      const fakeStrapi = {
+        getModel: jest.fn((modelName) => fakeModels[modelName]),
+      };
+
+      instance = createEntityService({
+        strapi: fakeStrapi,
+        db: fakeDB,
+        eventHub: new EventEmitter(),
+        entityValidator,
+      });
+    });
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+    afterAll(() => {
+      global.strapi.getModel.mockImplementation(() => ({}));
```

Comment:
Why is it needed to redefined the mock after all?

Reply:
I always cleanup the mock I use when my tests are done. I think it's a good practice to avoid surprises like writing a new test and found that `strapi.getModel('test-model')` return something but not the data I was expecting because it was set in another test above.

Reply:
I don't know a lot about mocking. Is there a way to remove the mock entirely instead of replacing by an empty function? clearAllMocks() does that ?

Reply:
Of course we can but as the `global.strapi.getModel()`method is mocked for the whole test file, I have just reseted it to its former mock declaration ( https://github.com/strapi/strapi/pull/14724/files#diff-ced24aa36409711f16daac290fcf9dc083f8d1431e2dc71418a0f034e79e546aR13 ).
If I remove entirely the mock implementation, it will just not return an empty object as describe by the first mock declaration.